design/KEPS/multicluster/proposal.md

```diff
@@ -0,0 +1,160 @@
+# Multicluster support
+
+1. [Summary](#summary)
+2. [Motivation](#motivation)
+   1. [Goals](#goals)
+   2. [Non-Goals](#nongoals)
+3. [Solution](#solution)
+   1. [Other solutions](#othersolutions)
+4. [Roadmap](#roadmap)
+
+# Summary
+
+Multi-cluster support for Kiali. A “single pane of glass” for your istio service mesh.
+
+[github discussions](https://github.com/kiali/kiali/discussions/categories/multicluster)
+
+# Motivation
+
+It is now common for meshes to span across multiple clusters and yet Kiali only partially supports multi-cluster deployments today. Users should be able to visualize their workloads on the graph across clusters, dig into details via the workload/app/service pages, receive validations for their istio config etc. across their whole mesh from a single Kiali.
+
+## Goals
+
+- Multi-cluster mesh support - single pane of glass for your entire mesh.
+- Support multi-primary and primary-remote Istio deployment models.
+- Support RBAC authentication strategy for multi-cluster deployments.
+
+## Non-goals
+
+- Custom aggregation of all Kiali’s datasources across clusters. There are other tools and initiatives for aggregating metrics/logs/traces such as otel and this does not need to be handled by Kiali.
+- Ambient is not going to be touched in this effort.
+- Multi-mesh support - multi-cluster is focused on single mesh support only but shouldn’t prevent Kiali from supporting multi-mesh in the future.
+
+# Solution
+
+A single Kiali configured to access each Kubernetes cluster that is part of the mesh. Kiali will require Metrics/traces to be aggregated in a centralized datastore that Kiali can query.
+
+In a multicluster deployment, Kiali will only support the OIDC auth strategy for now.
```

Comment:
Does this mean OpenShift will not be supported (i.e. `openshift` auth strategy)?

Reply:
nm.. I see it below... `The openshift strategy does not support multiple clusters.`

Reply:
There might be a way to configure your openshift clusters to use the same OIDC provider and for Kiali in turn to just use the `openshift` strategy but I'm not familiar enough with the `openshift` strategy to know if this would work without any changes to Kiali itself to support this. OIDC works out of the box today once you've configured Kiali and your clusters properly.